source/write/gridfs.txt

```diff
@@ -0,0 +1,389 @@
+         The default value is ``'fs'``.
+       | **Type**: ``string``
+
+   * - ``chunkSizeBytes`` 
+     - | Specifies the chunk size that GridFS splits files into. The default value is ``261120``.
+       | **Type**: ``integer``
+
+   * - ``readConcern`` 
+     - | Specifies the read concern to use for bucket operations. The default value is the 
+         database's read concern.
+       | **Type**: ``MongoDB\Driver\ReadConcern``
+
+   * - ``readPreference`` 
+     - | Specifies the read preference to use for bucket operations. The default value is the 
+         database's read preference.
+       | **Type**: ``MongoDB\Driver\ReadPreference``
+
+   * - ``writeConcern`` 
+     - | Specifies the write concern to use for bucket operations. The default value is the 
+         database's write concern.
+       | **Type**: ``MongoDB\Driver\WriteConcern``
+
+The following example creates a bucket named ``'myCustomBucket'`` by passing an
+array to ``selectGridFSBucket()`` that sets the ``bucketName`` option:
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start-create-custom-bucket
+   :end-before: end-create-custom-bucket
+
+.. _gridfs-upload-files:
+
+Upload Files
+------------
+
+You can upload files to a GridFS bucket by using the following methods:
+
+- ``MongoDB\GridFS\Bucket::openUploadStream()``: Opens a new upload stream
+  to which you can write file contents
+- ``MongoDB\GridFS\Bucket::uploadFromStream()``: Uploads the contents of
+  an existing stream to a GridFS file
+
+Write to an Upload Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``openUploadStream()`` method to create an upload stream for a given
+file name. The ``openUploadStream()`` method allows you to specify configuration
+information in an options array, which you can pass as a parameter.
+
+This example uses an upload stream to perform the following
+actions:
+
+- Opens a writable stream for a new GridFS file named ``'my_file'``
+- Sets the ``metadata`` option in an array parameter
+  to the ``openUploadStream()`` method
+- Calls the ``fwrite()`` method to write data to ``'my_file'``, which the stream points to
+- Calls the ``fclose()`` method to close the stream pointing to ``'my_file'`` 
+
+.. literalinclude:: /includes/write/gridfs.php
+   :language: php
+   :dedent:
+   :start-after: start-open-upload-stream
+   :end-before: end-open-upload-stream
+
+Upload an Existing Stream
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Use the ``uploadFromStream()`` method to upload the contents of a stream to
+a new GridFS file. The ``uploadFromStream()`` method allows you to specify configuration
+information in an options array, which you can pass as a parameter.
+
+This example performs the following actions:
+
+- Calls the ``fopen()`` method to create a temporary memory-based stream
+  that you can write to and read from
+- Calls the ``fwrite()`` method to write data to the stream
```

Comment:
If you take my suggestion on the example code, these can be reduced to a single bullet that calls `fopen()` on a local file. You can also note that we open the file in "binary read" mode (i.e. `rb`).

For the download example, we would `fopen()` or destination in "binary write" mode (i.e. `wb`).